fortitude/src/check.rs

```diff
@@ -519,6 +519,18 @@ pub(crate) fn check_only_file(
     let tree = parser
         .parse(file.source_text(), None)
         .context("Failed to parse")?;
+
+    let root_node = tree.root_node();
+    if let Some(rules) = ast_entrypoints.get(root_node.kind()) {
+        for rule in rules {
+            if let Some(violation) = rule.check(settings, &root_node, file) {
+                for v in violation {
+                    violations.push(v);
+                }
+            }
+        }
+    }
+
     for node in tree.root_node().named_descendants() {
```

Comment:
We can simplify this to:

```suggestion
    for node in [tree.root_node()].into_iter().chain(tree.root_node().named_descendants()) {
```

Only a little ugly!

Reply:
You can get it a little prettier with `std::iter::once`:

```rust
    let root = tree.root_node();
    for node in once(root).chain(root.named_descendants()) {
```